### PR TITLE
Avoiding unnecessary roundtrips to the database when evaluating permissions

### LINKED ISSUE #12148: Avoiding unnecessary roundtrips to the database when evaluating permissions
### Describe the bug

The recent changes to refactor the authorization store introduced a performance degradation when evaluating permissions. Basically, unnecessary database queries are being executed even though the data being queried is available in the cache.

There is also an unnecessary query being executed when querying for client scopes where if the scope is the client itself we should just return the client instead of executing a database query.

### Version

17.0.1

### Expected behavior

When evaluating permissions database queries should not happen if the cache is hot.

### Actual behavior

Unnecessary queries are being run even though the cache is hot.

### How to Reproduce?

* Enable logging to log database queries
* Create a client and enable authorization services
* Request an access token using the `urn:ietf:params:oauth:grant-type:uma-ticket`

You should see unnecessary queries to fetch client scopes, resource server, and associated policies even though entries are cached after the first run.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by pedroigor**
Closes #12148

### CODE REVIEW

**Review by vramik — COMMENTED** [review]
Thanks @pedroigor. Changes looks good to me. I have one minor comment regarding the order of parameters in constructor. But as it touches only legacy infinispan I'll leave it up to you if you want to incorporate or not.

File: model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/authorization/ResourceServerAdapter.java [hunk 1]
```diff
@@ -32,11 +32,14 @@
 public class ResourceServerAdapter implements ResourceServer, CachedModel<ResourceServer> {
     protected CachedResourceServer cached;
     protected StoreFactoryCacheSession cacheSession;
+    private RealmModel realm;
     protected ResourceServer updated;
 
-    public ResourceServerAdapter(CachedResourceServer cached, StoreFactoryCacheSession cacheSession) {
+    public ResourceServerAdapter(CachedResourceServer cached, StoreFactoryCacheSession cacheSession,
+            RealmModel realm) {
```
Comment (vramik) [inline model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/authorization/ResourceServerAdapter.java:39]: In map storage, when using `RealmModel` as parameter, it's always as first parameter, So in sake of consistency:
```suggestion
    public ResourceServerAdapter(RealmModel realm, CachedResourceServer cached,
            StoreFactoryCacheSession cacheSession) {
```

**Review by mhajas — DISMISSED** [review]
Thanks for the fix @pedroigor

File: model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/authorization/ResourceServerAdapter.java (hunk 1 above)
Comment (hmlnarik) [reply model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/authorization/ResourceServerAdapter.java:39]: I would prefer the same

**Review by vramik — APPROVED** [review]

**Review by hmlnarik — APPROVED** [review]

**Pull request merged**